subprojects/docs/src/docs/userguide/running-builds/configuration_cache.adoc

```diff
@@ -359,119 +580,149 @@ For example if you need a `Logger` in your task actions you should use `Task.log
+
+As you can see above, fixing ad-hoc tasks declared in scripts requires quite a bit of ceremony.
+It is a good time to think about extracting your task declaration as a proper task class as shown previously.
+
+The following table shows what APIs or injected service should be used as a replacement for each of the `Project` methods.
 
 [cols="a,a",options="header"]
 |===
 | Instead of:
 | Use:
 
 | `project.rootDir`
-| A task field or script variable to capture the root directory.
+| A task property or script variable to capture the root directory.
 
 | `project.projectDir`
-| `link:{javadocPath}/org/gradle/api/file/ProjectLayout.html[ProjectLayout].projectDirectory`
+| link:{javadocPath}/org/gradle/api/file/ProjectLayout.html#getProjectDirectory--[ProjectLayout.projectDirectory]
 
 | `project.buildDir`
-| `link:{javadocPath}/org/gradle/api/file/ProjectLayout.html[ProjectLayout].buildDirectory`
+| link:{javadocPath}/org/gradle/api/file/ProjectLayout.html#getBuildDirectory--[ProjectLayout.buildDirectory]
 
 | `project.name`
-| A task field or script variable to capture the project name.
+| A task property or script variable to capture the project name.
 
 | `project.description`
-| A task field or script variable to capture the project description.
+| A task property or script variable to capture the project description.
 
 | `project.group`
-| A task field or script variable to capture the project group.
+| A task property or script variable to capture the project group.
 
 | `project.version`
-| A task field or script variable to capture the project version.
+| A task property or script variable to capture the project version.
 
 | `project.properties`, `project.property(name)`, `project.hasProperty(name)`, `project.getProperty(name)` or `project.findProperty(name)`
 | <<configuration_cache#undeclared_gradle_prop_read, Value providers for Gradle properties>>
 
+| `project.logger`
+| link:{javadocPath}/org/gradle/api/Task.html#getLogger--[Task.logger]
+
 | `project.provider {}`
-| `link:{javadocPath}/org/gradle/api/provider/ProviderFactory.html[ProviderFactory].provider {}`
+| link:{javadocPath}/org/gradle/api/provider/ProviderFactory.html#provider-java.util.concurrent.Callable-[ProviderFactory.provider {}]
 
 | `project.file(path)`
-| `link:{javadocPath}/org/gradle/api/file/ProjectLayout.html[ProjectLayout].projectDirectory.file(path)` or `link:{javadocPath}/org/gradle/api/file/ProjectLayout.html[ProjectLayout].buildDirectory.file(path)`
+| link:{javadocPath}/org/gradle/api/file/ProjectLayout.html#getProjectDirectory--[ProjectLayout.projectDirectory].link:{javadocPath}/org/gradle/api/file/Directory.html#file-java.lang.String-[file(path)] or link:{javadocPath}/org/gradle/api/file/ProjectLayout.html#getBuildDirectory--[ProjectLayout.buildDirectory].link:{javadocPath}/org/gradle/api/file/DirectoryProperty.html#file-java.lang.String-[file(path)]
 
 | `project.uri(path)`
-| [.red]#✖# Unsupported
+| [.red]#✖# Not supported yet
```

Comment:
It may make sense to pass this in to the task via a property, if it represents a parameter of the task action. Otherwise, File.toURI() or some other JVM API can be used